app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt

```diff
@@ -0,0 +1,12 @@
+package org.fossasia.openevent.general.discount
+
+import io.reactivex.Single
+import org.fossasia.openevent.general.auth.DiscountCode
+import retrofit2.http.GET
+
+interface DiscountApi {
+
+    @GET("discount-codes?size=0")
```

Comment:
page[size]=0

Reply:
Also, discount codes should be loaded on per event basis

Reply:
That means that there is only one discount code per event

About loading every discount code, in production code, there can be 1000s of events, not a good idea. Request will time out

Reply:
the responses will be paged but, right? that means loading only fixed no of discount codes on every fetch

Reply:
But then what'd be the point, the discount codes will be out of sync with the loaded events. You won't even be able to show which event discount code belongs to

Reply:
got the point! what should be done then?

Reply:
Load the discount code when the event is clicked

Reply:
okay!